activemodel/test/cases/type/big_integer_test.rb

```diff
@@ -21,6 +21,12 @@ def test_large_values
         assert_equal 9999999999999999999999999999999, type.serialize(9999999999999999999999999999999)
       end
 
+      def test_large_integers_are_not_out_of_range
+        type = Type::BigInteger.new
+
+        assert_equal false, type.send(:out_of_range?, 2**63)
```

Comment:
I don't think this is testing much. BigDecimal overwrites almost all methods from integer where `out_of_range?` would be used:
```rb
type = ActiveModel::Type::BigInteger.new
type.serializable?(2**128)
type.serialize(2**128)
type.serialize("340282366920938463463374607431768211456")
```

This all work as expected, even though the number is very large. Is there some other test you can write? I think there may be an issue when it calls into `super` from `serialize`, maybe it misbehaves for number-like custom types?

Reply:
Good catch, yea its number-like custom classes where this breaks, let me try and write a test.